### PR TITLE
For issue #8570: Disabling uBlock and quickly opening the Settings section cause a crash

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.
 
### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/addons/InstalledAddonDetailsFragment.kt [hunk 1]
```diff
@@ -94,7 +94,7 @@ class InstalledAddonDetailsFragment : Fragment() {
         switch.setOnCheckedChangeListener { v, isChecked ->
             val addonManager = v.context.components.addonManager
             switch.isClickable = false
-            remove_add_on.isEnabled = false
+            view.remove_add_on.isEnabled = false
```
Comment (psymoon) [inline app/src/main/java/org/mozilla/fenix/addons/InstalledAddonDetailsFragment.kt]: Interesting. Does kotlin view binding not recognize/handle the differences here?

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/addons/InstalledAddonDetailsFragment.kt]: It looks like the Kotlin synthetic view binding variable get null at some point when we are switching between screen and as this code gets called async the null pointer exception happens

File: app/src/main/java/org/mozilla/fenix/addons/InstalledAddonDetailsFragment.kt [hunk 2]
```diff
@@ -129,14 +128,14 @@ class InstalledAddonDetailsFragment : Fragment() {
                     }
                 )
             } else {
+                view.settings.isVisible = false
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/addons/InstalledAddonDetailsFragment.kt]: I had to add this extra step here, as we are hiding the settings section after  `onSuccess` if the user clicks fast enough can enter to the setting page, but when the user arrives there the add-on is already disabled and the page can't be shown.

image

**Comment by gabrielluong** [comment]
Can you also check if we need to make this change in r-b and sample-browser

**Comment by Amejia481** [comment]
Sure!

**Pull request merged**